### PR TITLE
Added rotateAround method to Vector2

### PR DESCRIPTION
**Pull request by please-wait**
rotates the Vector2, around another 2D coordinate (e.g. another Vector2).

### CODE REVIEW

**Comment by mrdoob** [comment]
@WestLangley do you think we should include this?

**Comment by WestLangley** [comment]
@please-wait What is your use-case?

@mrdoob So far, there has been no demand for such a method. Let's see if there is a compelling use case.

In any event, if it were included, I would rewrite it like so:

```javascript
rotateAround: function ( center, angle ) {

	var c = Math.cos( angle ), s = Math.sin( angle );

	var x = this.x - center.x;
	var y = this.y - center.y;

	this.x = x * c - y * s + center.x;
	this.y = x * s + y * c + center.y;

	return this;

}
```

**Comment by please-wait** [comment]
@WestLangley, long story short, [when you google](https://www.google.ca/search?q=rotate%202d%20point%20around%20another%20point), you see many game developers asking about how to do this in Unity so it has general uses for video games (e.g. determining player's action based on the analog stick movements).

>So far, there has been no demand for such a method.

Then I think I am going to be the first one :D
 I used to make a Vector3 and use ApplyAxisAngle myself, that works fine, but thought this would make more sense.

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#7307

**Comment by EliasHasle** [comment]
Would it be even better with:

```
rotate: function ( angle, center ) {

    if ( center === undefined) {

        center = _v.set( 0, 0 ); //where _v is preallocated in module scope

    }

    var c = Math.cos( angle ), s = Math.sin( angle );

    var x = this.x - center.x;
    var y = this.y - center.y;

    this.x = x * c - y * s + center.x;
    this.y = x * s + y * c + center.y;

    return this;

}
```
?
TL;DR: `rotate` with mandatory `angle` and optional `center` defaulting to the origin (i.e. treating the vector as a direction). This is what I repeatedly see myself expecting to find in `Vector2`.

**Comment by mrdoob** [comment]
@EliasHasle Would you like to create a new issue for it?

**Comment by EliasHasle** [comment]
OK. Done.
